### PR TITLE
[Feature](function) Support width_bucket function

### LINKED ISSUE #13981: WIDTH_BUCKET function support
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

> function **width_bucket** had supported by oracle and postgresql,hive,sparksql,impala,snowflake and  databricks.
please add the same name function and have the same feature.
please refer to:
https://www.postgresql.org/docs/15/functions-math.html
https://docs.snowflake.com/en/sql-reference/functions/width_bucket.html

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Yukang-Lian**
# Proposed changes

Issue Number: close #13981 

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.78 seconds
 load time: 443 seconds
 storage size: 17150166363 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221118125919_clickbench_pr_48778.html

File: docs/zh-CN/docs/sql-manual/sql-functions/width-bucket.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+#### Syntax
+
+```sql
+width_bucket(expr, min_value, max_value, num_buckets)
+```
+
+#### Arguments
+`expr` -
+创建直方图的表达式。此表达式必须计算为数值或可隐式转换为数值的值。
```
Comment (morningman) [inline docs/zh-CN/docs/sql-manual/sql-functions/width-bucket.md:41]: 这里说必须是数值，但是下面的示例中，可以使用 `date` 类型？

File: be/src/vec/functions/function_width_bucket.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                nested_column_concrete.get_data()[i] = num_buckets + 1;
+            } else {
+                if ((max_value - min_value) / num_buckets == 0) {
+                    continue;
+                }
+                nested_column_concrete.get_data()[i] =
+                        (int64_t)(1 + (expr_column_concrete.get_data()[i] - min_value) /
+                                              ((max_value - min_value) / num_buckets));
+            }
+            LOG(WARNING) << std::to_string(i) + "   " +
```
Comment (morningman) [inline be/src/vec/functions/function_width_bucket.cpp:106]: Do not print WARNING log here

File: docs/zh-CN/docs/sql-manual/sql-functions/width-bucket.md (hunk 1 above)
Comment (Yukang-Lian) [reply docs/zh-CN/docs/sql-manual/sql-functions/width-bucket.md:41]: `date` can be implicitly converted to a numeric value. Actually, `date` is `Int64`. https://github.com/apache/doris/blob/ff197b0fa537e076ff1a0b5c43437552fb6d8ba1/be/src/vec/core/types.h#L262-L265

File: be/src/vec/functions/function_width_bucket.cpp (hunk 2 above)
Comment (Yukang-Lian) [reply be/src/vec/functions/function_width_bucket.cpp:106]: This is used for debugging. I forgot to remove it.

File: be/src/vec/functions/function_width_bucket.cpp [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        int64_t num_buckets = num_buckets_ptr->get_int(0);
+
+        auto nested_column_ptr = ColumnInt64::create(input_rows_count, 0);
+        DataTypePtr expr_type = block.get_by_position(arguments[0]).type;
+
+        _execute_by_type(*expr_ptr, *min_value_ptr, *max_value_ptr, num_buckets, *nested_column_ptr,
+                         expr_type);
+
+        WhichDataType which(remove_nullable(expr_type));
+        if (which.is_nullable()) {
```
Comment (zhangstar333) [inline be/src/vec/functions/function_width_bucket.cpp:65]: see have remove nullable in 64 line?

**Review by zhangstar333 — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**